src/battle_util.c

```diff
@@ -360,6 +360,7 @@ void HandleAction_UseMove(void)
                  || (GetBattlerAbility(battler) == ABILITY_STORM_DRAIN && moveType == TYPE_WATER))
                 && GetBattlerTurnOrderNum(battler) < var
                 && gBattleMoves[gCurrentMove].effect != EFFECT_SNIPE_SHOT
+                && gBattleMoves[gCurrentMove].effect != EFFECT_PLEDGE
```

Comment:
Storm Drain would still redirect a combined Water Pledge. This may need some testing.

Reply:
On [showdown](https://replay.pokemonshowdown.com/gen9doublesubers-1963224510) it doesn't.